slackbridge/bots.py

```diff
@@ -198,28 +199,74 @@ class UserBot(IRCBot):
     def __init__(self, nickname, realname, user_id, channels,
                  target_group, nickserv_pw):
         self.nickname = '{}-slack'.format(utils.strip_nick(nickname))
+        self.intended_nickname = self.nickname
         self.realname = realname
         self.user_id = user_id
         self.channels = channels
         self.nickserv_password = nickserv_pw
         self.target_group_nick = target_group
 
+        self.nick_change_loop = LoopingCall(
+            self.setNick, self.intended_nickname)
```

Comment:
I'd format this as the following, to avoid any confusion and allow for a trailing comma (I mention this a few times in this review, just as nitpicks though, nothing major).

    self.nick_change_loop = LoopingCall(
        self.setNick,
        self.intended_nickname,
    )

Reply:
I think I mentioned this months ago on IRC, but have you tried doing this without a ```LoopingCall```? From my understanding ```irc_ERR_NICKNAMEINUSE``` will be called anytime the attempted nick is in use. So whenever a nickserv auth fails you can trust that either ```irc_ERRNICKNAMEINUSE``` and/or ```nickChanged``` will be called. Those methods plus ```reactor.callLater``` can create a more natural event based loop that will stop once the desired nick is in use.